pkg/util/validation/limits.go

```diff
@@ -26,30 +26,31 @@ import (
-	creationGracePeriodFlag                = "validation.create-grace-period"
-	maxPartialQueryLengthFlag              = "querier.max-partial-query-length"
-	maxTotalQueryLengthFlag                = "query-frontend.max-total-query-length"
-	maxQueryExpressionSizeBytesFlag        = "query-frontend.max-query-expression-size-bytes"
-	requestRateFlag                        = "distributor.request-rate-limit"
-	requestBurstSizeFlag                   = "distributor.request-burst-size"
-	ingestionRateFlag                      = "distributor.ingestion-rate-limit"
-	ingestionBurstSizeFlag                 = "distributor.ingestion-burst-size"
-	HATrackerMaxClustersFlag               = "distributor.ha-tracker.max-clusters"
-	resultsCacheTTLFlag                    = "query-frontend.results-cache-ttl"
-	resultsCacheTTLForOutOfOrderWindowFlag = "query-frontend.results-cache-ttl-for-out-of-order-time-window"
-	QueryIngestersWithinFlag               = "querier.query-ingesters-within"
+	MaxSeriesPerMetricFlag                   = "ingester.max-global-series-per-metric"
+	MaxMetadataPerMetricFlag                 = "ingester.max-global-metadata-per-metric"
+	MaxSeriesPerUserFlag                     = "ingester.max-global-series-per-user"
+	MaxMetadataPerUserFlag                   = "ingester.max-global-metadata-per-user"
+	MaxChunksPerQueryFlag                    = "querier.max-fetched-chunks-per-query"
+	MaxChunkBytesPerQueryFlag                = "querier.max-fetched-chunk-bytes-per-query"
+	MaxSeriesPerQueryFlag                    = "querier.max-fetched-series-per-query"
+	maxLabelNamesPerSeriesFlag               = "validation.max-label-names-per-series"
+	maxLabelNameLengthFlag                   = "validation.max-length-label-name"
+	maxLabelValueLengthFlag                  = "validation.max-length-label-value"
+	maxMetadataLengthFlag                    = "validation.max-metadata-length"
+	maxNativeHistogramBucketsFlag            = "validation.max-native-histogram-buckets"
+	creationGracePeriodFlag                  = "validation.create-grace-period"
+	maxPartialQueryLengthFlag                = "querier.max-partial-query-length"
+	maxTotalQueryLengthFlag                  = "query-frontend.max-total-query-length"
+	maxQueryExpressionSizeBytesFlag          = "query-frontend.max-query-expression-size-bytes"
+	requestRateFlag                          = "distributor.request-rate-limit"
+	requestBurstSizeFlag                     = "distributor.request-burst-size"
+	ingestionRateFlag                        = "distributor.ingestion-rate-limit"
+	ingestionBurstSizeFlag                   = "distributor.ingestion-burst-size"
+	HATrackerMaxClustersFlag                 = "distributor.ha-tracker.max-clusters"
+	resultsCacheTTLFlag                      = "query-frontend.results-cache-ttl"
+	resultsCacheTTLForOutOfOrderWindowFlag   = "query-frontend.results-cache-ttl-for-out-of-order-time-window"
+	resultsCacheForUnalignedQueryEnabledFlag = "query-frontend.cache-unaligned-requests"
```

Comment:
[nit] No need for this constant, given it's just referenced once. Just replace the constant with the actual string below.